### PR TITLE
metrics: optimize prometheus metrics label

### LINKED ISSUE #2183: Metrics labels should not contain ID, table name or other unbounded sets of values
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

Running a changefeed which captures 20k tables produces large metric records.

We should not put ID and table names in metrics labels.

> CAUTION: Remember that every unique combination of key-value label pairs represents a new time series, which can dramatically increase the amount of data stored. Do not use labels to store dimensions with high cardinality (many different label values), such as user IDs, email addresses, or other unbounded sets of values.
>
> https://prometheus.io/docs/practices/naming/#labels

Here is a list of problematic metrics

- [x] ticdc_kvclient_region_token
 * not monitored
- [x] ticdc_sorter_consume_count
 *  no need to update
- [x] ticdc_sorter_event_count
 * no need to update
- [x] ticdc_sorter_flush_count_histogram
 * no need to update
- [x] ticdc_sorter_merge_count_histogram
 * no need to update
- [x] ticdc_sorter_merger_start_ts_gauge
 * not monitored
- [x] ticdc_sorter_resolved_ts_gauge
 * not monitored
- [x] ticdc_kvclient_channel_size
 * not monitored
- [x] ticdc_processor_table_memory_consumption
   - [x] histogram
   - [x] update grafana
     - [x] Processor Memory Consumption Per Capture
     - [x] Processor Memory Consumption Per Table -> Per Capture 
- [x] ticdc_processor_table_resolved_ts
  - [x] Table Resolved ts 
- [x] ticdc_puller_event_chan_size
   - [x] histogram
   - [x] update grafana 
     - [x] Puller buffer size
- [x] ticdc_puller_output_chan_size
   - [x] histogram
   - [x] update grafana 
     - [x] Puller buffer size 
- [x] ticdc_puller_resolved_ts
 * not monitored
- [x] ticdc_puller_txn_collect_event_count
 * no need to update

**Any unbounded sets of labels should be avoided, and use other ways to record critical metrics.**

For above metrics, we should

* Remove the `table` and other labels that have high cardinality.
* Use sampling.

Eg. we can rewrite `ticdc_processor_table_memory_consumption` to 

```go
// Sampling, turn gague/counter into histogram.
tableMemoryGauge = prometheus.NewGaugeVec(
	prometheus.GaugeOpts{
		Namespace: "ticdc",
		Subsystem: "processor",
		Name:      "table_memory_consumption",
		Help:      "estimated memory consumption for a table after the sorter",
	}, []string{"changefeed", "capture", "table"})

tableMemoryHistogram = prometheus.NewHistogramVec(
	prometheus.HistogramOpts{
		Namespace: "ticdc",
		Subsystem: "processor",
		Name:      "table_memory_consumption",
		Help:      "estimated memory consumption for a table after the sorter",
	}, []string{"changefeed", "capture"})
```

2. What did you expect to see?

A smaller metric record.

3. What did you see instead?

A large metrics records, 28MB. [metrics.tar.gz](https://github.com/pingcap/ticdc/files/6730807/metrics.tar.gz)

4. Versions of the cluster

v5.0.2/v4.0.13

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve? 
Try to close #2183 

### What is changed and how it works?

* since grafana already updated to 7.5, this PR has some related metadata change
* remove label `table` in some metrics, as mentioned in the #2183 
* use the histogram to replace gauge for some metrics, and update related grafana display, this would cause the figure not the same before.

### Check List 

Tests 

 - No code

Code changes

Side effects

Related changes

### Release note 

```release-note
 `None`.
```

### CODE REVIEW

**Comment by 3AceShowHand** [comment]
Screen Shot 2021-08-06 at 22 52 49
* Processor Memory Consumption Per Capture
  * "sum(rate(ticdc_processor_table_memory_consumption_sum{tidb_cluster=\"$tidb_cluster\", capture=~\"$capture\"}[30s]) / rate(ticdc_processor_table_memory_consumption_count{tidb_cluster=\"$tidb_cluster\", capture=~\"$capture\"}[30s])) by (capture)"
* Processor Memory Consumption Per Changefeed
  * "sum(rate(ticdc_processor_table_memory_consumption_sum{tidb_cluster=\"$tidb_cluster\", capture=~\"$capture\"}[30s]) / rate(ticdc_processor_table_memory_consumption_count{tidb_cluster=\"$tidb_cluster\", capture=~\"$capture\"}[30s])) by (capture, capture)"

**Comment by 3AceShowHand** [comment]
Screen Shot 2021-08-06 at 23 23 26

**Comment by 3AceShowHand** [comment]
For review, just ignore unrelated changes, focus on metrics logic, and grafana display related things, other meta info will be updatedd after passing the review, before the code merge.

**Comment by 3AceShowHand** [comment]
For review, please help me verify the correctness of Grafana PromQL

**Comment by 3AceShowHand** [comment]
image

**Review by overvenus — APPROVED** [review]
LGTM

File: metrics/grafana/ticdc.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "__inputs": [
     {
-      "name": "DS_TEST-CLUSTER",
-      "label": "test-cluster",
+      "name": "DS_LINGJIN-TEST",
+      "label": "lingjin-test",
```
Comment (amyangfei) [inline metrics/grafana/ticdc.json:5]: please revert this change, ditto for the following names

File: metrics/grafana/ticdc.json [hunk 2]
```diff
@@ -14,7 +14,7 @@
       "type": "grafana",
       "id": "grafana",
       "name": "Grafana",
-      "version": "6.1.6"
+      "version": "7.5.7"
```
Comment (amyangfei) [inline metrics/grafana/ticdc.json:17]: Whether Grafana in all release branches have been upgraded?

File: metrics/grafana/ticdc.json [hunk 3]
```diff
@@ -7846,15 +8154,20 @@
...
         "hide": 0,
         "includeAll": true,
         "label": "Runtime metrics",
         "multi": false,
         "name": "runtime_instance",
         "options": [],
-        "query": "label_values(process_start_time_seconds{tidb_cluster=\"$tidb_cluster\", job=\"ticdc\"}, instance)",
+        "query": {
+          "query": "label_values(process_start_time_seconds{tidb_cluster=\"$tidb_cluster\", job=\"ticdc\"}, instance)",
+          "refId": "lingjin-test-runtime_instance-Variable-Query"
```
Comment (amyangfei) [inline metrics/grafana/ticdc.json:8169]: Please update the refId

File: metrics/grafana/ticdc.json [hunk 4]
```diff
@@ -7897,7 +8210,7 @@
     ]
   },
   "timezone": "browser",
-  "title": "Test-Cluster-TiCDC",
-  "uid": "YiGL8hBZ1",
-  "version": 18
+  "title": "lingjin-test-TiCDCxx",
+  "uid": "YiGL8hBZ1xxx",
+  "version": 1
```
Comment (amyangfei) [inline metrics/grafana/ticdc.json:8215]: Please revert this change, and better to increase version

File: metrics/grafana/ticdc.json (hunk 2 above)
Comment (3AceShowHand) [reply metrics/grafana/ticdc.json:17]: yes, all branches affected.

File: metrics/grafana/ticdc.json [hunk 5]
```diff
@@ -36,8 +36,14 @@
     },
     {
       "type": "panel",
-      "id": "table",
-      "name": "Table",
+      "id": "table-old",
+      "name": "Table (old)",
```
[the remark this answers is not in the document]
Comment (3AceShowHand) [reply metrics/grafana/ticdc.json]: Since Grafana 7.5 introduce a new table, so old table become `table-old` in meta

File: metrics/grafana/ticdc.json (hunk 3 above)
Comment (3AceShowHand) [reply metrics/grafana/ticdc.json]: rename `lingjin-test` -> `test-cluster`

**Comment by amyangfei** [comment]
Still can't import in Grafana v6.1.6, with following error

```
Templating init failed
e.replace is not a function
```

**Comment by 3AceShowHand** [comment]
> Still can't import in Grafana v6.1.6, with following error
> 
> ```
> Templating init failed
> e.replace is not a function
> ```


It looks like a Grafana's bug, I will fix this asap.

**Review by amyangfei — APPROVED** [review]
